tests/helpers/test_config_validation.py

```diff
@@ -712,6 +712,30 @@ def test_deprecated_with_no_optionals(caplog, schema):
     assert test_data == output
 
 
+def test_deprecated_removed_param(caplog, schema):
+    """
+    Test removed deprecation and fail the config validation.
+
+    Expected behavior:
+        - Outputs the appropriate deprecation error if key is detected and removed from support
+    """
+    deprecated_schema = vol.All(cv.removed("mars"), schema)
+
+    test_data = {"mars": True}
+    with pytest.raises(vol.Invalid) as excinfo:
+        deprecated_schema(test_data)
+    assert (
+        "The 'mars' option is deprecated, please remove it from your configuration"
```

Comment:
This is a tiny bit weird... the method is for a removed parameter, the message tells if has been deprecated... 

Is it removed or deprecated?

Reply:
Actually both are true. cv.removed handles removed deprecated options by failing the setup. cv.deprecated only logs a warning.

Reply:
removed != deprecated.
So both cannot be true.

Reply:
The current code shares the the formatting code for deprecated or removed options.

Reply:
Solved